src/java.base/share/classes/sun/launcher/LauncherHelper.java

```diff
@@ -632,6 +633,13 @@ static String getMainClassFromJar(String jarname) {
             if (opens != null) {
                 addExportsOrOpens(opens, true);
             }
+            String enableNativeAccess = mainAttrs.getValue(ENABLE_NATIVE_ACCESS);
+            if (enableNativeAccess != null) {
+                if (!enableNativeAccess.equals("ALL-UNNAMED")) {
+                    throw new IllegalArgumentException("Only ALL-UNNAMED allowed as value for " + ENABLE_NATIVE_ACCESS);
+                }
```

Comment:
I don't think throwing IAE is right here. It should call abort with a key for the error message. The value of enableNativeAccess can be used as the parameter for the message.